### PR TITLE
Bump deno version

### PR DESCRIPTION
**Pull request by kylewillmon**
This includes migration to the new `op2` macro because `op` is now deprecated (See denoland/deno_core#279). The new macro requires explicit annotation for string and serde parameters and return values because of the performance hit. We use these heavily (and don't care about the performance).

### CODE REVIEW

File: cli/src/commands/extensions/api.rs [hunk 1]
```diff
@@ -496,19 +514,22 @@ fn add_permission_args<'a>(
 }
 
 /// Return error when trying to sandbox on Windows.
-#[op]
 #[cfg(not(unix))]
-fn run_sandboxed(_process: Process) -> Result<ProcessOutput> {
+#[op2]
+#[serde]
+fn run_sandboxed(#[serde] _process: Process) -> Result<ProcessOutput> {
```
Comment (kylewillmon) [inline cli/src/commands/extensions/api.rs:520]: Just have to hope I did this one right... We don't test native Windows builds anymore in CI (and I don't have a Windows host to test on)...

Maybe I can convince @furi0us333 to test this for me... Or we should discuss using [windows-xl][1] runners (at cost) periodically to test this sort of stuff.

[1]: https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners

**Comment by kylewillmon** [comment]
Looks like the build requires `protoc` now, which is causing CI to fail....

I'll investigate later today to see if I can avoid the dependency

File: cli/src/commands/extensions/api.rs (hunk 1 above)
Comment (furi0us333) [reply cli/src/commands/extensions/api.rs:520]: I am happy to test something if needed, though I always try and push people to using WSL.

Comment (kylewillmon) [reply cli/src/commands/extensions/api.rs:520]: Me too. We don't officially support Windows outside of WSL.

But we've historically made an attempt to at least build successfully on native Windows. So I was just curious if that still worked or not...

**Comment by kylewillmon** [comment]
It looks like there is no opt out. `protoc` has been required to build deno since denoland/deno@2d9298f

I'll update the workflows to restore CI.

**Review by cd-work — DISMISSED** [review]

**Review by cd-work — APPROVED** [review]

File: Cargo.lock [hunk 2]
```diff
@@ -4418,6 +4471,22 @@ version = "1.0.15"
...
+
+[[package]]
+name = "saffron"
+version = "0.1.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "03fb9a628596fc7590eb7edbf7b0613287be78df107f5f97b118aad59fb2eea9"
+dependencies = [
+ "chrono",
+ "nom 5.1.3",
+]
```
Comment (kylewillmon) [inline Cargo.lock:4488]: This project is (basically) dead, but [deno is aware][1]. Hopefully, they'll replace it or get PRs merged to update it soon.

[1]: https://github.com/denoland/deno/pull/21019#discussion_r1377019647

**Comment by kylewillmon** [comment]
I've opened littledivy/aead-gcm-stream#2 to fix the build error on Rust 1.71.0

Hoping to get a quick response there. I'd rather not bump MSRV beyond the latest 3 minor releases.

**Review by cd-work — APPROVED** [review]

File: cli/src/deno.rs [hunk 3]
```diff
@@ -49,12 +49,8 @@ pub async fn run(
     let main_module =
         deno_core::resolve_path(&extension.entry_point().to_string_lossy(), &PathBuf::from("."))?;
 
-    let bootstrap = BootstrapOptions {
-        args,
-        runtime_version: env!("CARGO_PKG_VERSION").into(),
```
Comment (kylewillmon) [inline cli/src/deno.rs:54]: This was putting the Phylum CLI version in `Deno.version.deno`. That variable is now blank (set [here][1] in the `deno_runtime` snapshot).

[1]: https://github.com/denoland/deno/blob/9444bab18647bc871b63716a3a70b2aebaaf3821/runtime/build.rs#L38-L40

We could consider injecting the Phylum CLI version into the extension runtime somewhere else. But that would be a separate PR.
